packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -920,14 +923,20 @@ export class BlobManager extends TypedEventEmitter<IBlobManagerEvents> {
 				}
 				const blobs = {};
 				const localBlobs = new Set<PendingBlob>();
+				const abortPromise = new Promise<void>((resolve, reject) => {
+					stopBlobAttachingSignal?.addEventListener(
+						"abort",
+						() => {
+							reject(new Error("Operation aborted"));
+						},
+						{ once: true },
+					);
+				});
 				while (localBlobs.size < this.pendingBlobs.size) {
 					const attachBlobsP: Promise<void>[] = [];
 					for (const [id, entry] of this.pendingBlobs) {
 						if (!localBlobs.has(entry)) {
 							localBlobs.add(entry);
-							if (!entry.opsent) {
-								this.sendBlobAttachOp(id, entry.storageId);
-							}
 							entry.handleP.resolve(this.getBlobHandle(id));
 							attachBlobsP.push(
```

Comment:
>>rather than being something we only monitor externally to the individual blob attaches, it might be better to integrate it inside such that we reject/resolve the blobP for any incomplete blobs on cancelation.

handle the abort inside this promise, so we actually stop doing work, rather than aborting the waiting